src/main/java/org/ehrbase/fhirbridge/ehr/converter/specific/virologischerbefund/ProAnalytZugehoerigeLaborprobeChoiceConverter.java

```diff
@@ -0,0 +1,93 @@
+package org.ehrbase.fhirbridge.ehr.converter.specific.virologischerbefund;
+
+import com.nedap.archie.rm.datavalues.DvIdentifier;
+import org.ehrbase.fhirbridge.ehr.opt.virologischerbefundcomposition.definition.ProAnalytZugehoerigeLaborprobeChoice;
+import org.ehrbase.fhirbridge.ehr.opt.virologischerbefundcomposition.definition.ProAnalytZugehoerigeLaborprobeDvUri;
+import org.ehrbase.fhirbridge.ehr.opt.virologischerbefundcomposition.definition.ProAnalytZugehoerigeLaborprobeDvIdentifier;
+
+import org.hl7.fhir.r4.model.Observation;
+
+import java.util.Optional;
+
+public class ProAnalytZugehoerigeLaborprobeChoiceConverter {
+
+    public ProAnalytZugehoerigeLaborprobeChoice convertDvIdentifier(Observation observation){
+
+        ProAnalytZugehoerigeLaborprobeDvIdentifier proAnalytZugehoerigeLaborprobeDvIdentifier = new ProAnalytZugehoerigeLaborprobeDvIdentifier();
+
+        mapDvIdentifier(observation).ifPresent(proAnalytZugehoerigeLaborprobeDvIdentifier::setZugehoerigeLaborprobe);
+
+        return proAnalytZugehoerigeLaborprobeDvIdentifier;
+
+    }
+
+    public Optional<DvIdentifier> mapDvIdentifier(Observation observation){
+
+        DvIdentifier dvIdentifier = new DvIdentifier();
+
+        mapAssigner(observation).ifPresent(dvIdentifier::setAssigner);
+        mapId(observation).ifPresent(dvIdentifier::setId);
+        mapType(observation).ifPresent(dvIdentifier::setType);
+
+        return Optional.of(dvIdentifier);
```

Comment:
i would suggest to add here a hasIdentifier and if not an Optional.empty(); is to be returned (does this make sense? since i dont see no check for identifier).
If identifier is there map it and return, if not dont map it and do nothing (also then an Optional.empty() should be returned by convertDvIdentifier since nothing is mapped).
Furthermore all methods here are public, im pretty sure most of them should be private at least.

If you dont have an empty output you also dont need optional :D in general

Nevertheless, we do have parser for that see [here](https://github.com/ehrbase/fhir-bridge/blob/develop/src/main/java/org/ehrbase/fhirbridge/ehr/converter/generic/DvIdentifierParser.java), this one has some errors as i can see it, maybe you take your stuff (since its better then mine) and put it there, i missed out some checks as it seems. Just dont forget the hasIdentifier check for Optional.empty()

Reply:
There is a hasIdentifier and an Optional.empty, but that is handled before we go into this class+method.
Here is the place where it happens (private Optional<ProAnalytZugehoerigeLaborprobeChoice> mapZugehoerigeLaborprobe  in ProAnalytClusterConverter):
https://github.com/ehrbase/fhir-bridge/pull/392/files#diff-4d8c593b51ad51f7cbd0a2addd7fb06a59da59dc9c48d4cf9a9e1edb3fa9dc5e

As I first need to determine which of the 2 methods (convertDvIdentifier and convertDvUri) needs to be used to map the resource the Identifier check is made earlier.
If there is an Identifier --> convertDvIdentifier
If there is no Identifier but a TypeElement --> convertDvUri
If there is no Identifier or TypeElement then Optional.empty() will be triggered.

Methods have been made private though:
https://github.com/ehrbase/fhir-bridge/pull/392/files#diff-585a1cf0f329e2ab64afa38a03d4bfe341376a4b2b2937848f1326f68da104f9